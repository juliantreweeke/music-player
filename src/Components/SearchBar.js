import React from "react";
import styled from "styled-components";
import  SVG  from './SVG'

const Container= styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom:150px;
`;
const Searchbox = styled.div`
  position:relative;
  top:50px;
  background:black;
  height:60px;
  border-radius:40px;
  cursor:pointer;
  transition: width 2s;
`;

const SearchInput = styled.input`
  float:left;
  background:none;
  border:none;
  outline:none;
  color:rgb(112, 193, 179);
  font-size: 1em;
  transition: 0.4s;
  padding:0;
  height:60px;
  line-height:60px
  transition: width 0.8s;
  ${Searchbox}:hover & {
    width:200px;
  }
  text-indent: 2em;
  width:0px;
`

const SearchButton = styled.a`
  float:right;
  background: black;
  height:60px;
  width:60px;
  border-radius:50%;
  cursor:pointer;
  transition: background 0.8s;
  ${Searchbox}:hover & {
    background:white;
  }
`;

const SearchBar = ({
        handleInputChange,
        searchTracks
    }) => {
    return (
        <Container>
            <Searchbox>
                <SearchInput
                    placeholder="Search for..."
                    onChange={handleInputChange}
                /> 
                <SearchButton onClick={searchTracks}><SVG fill="rgba(112, 193, 179, 1)" /></SearchButton>
            </Searchbox>
        </Container>
    );
};

export default SearchBar;