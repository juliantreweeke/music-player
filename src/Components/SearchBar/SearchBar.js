import React from "react";
import styled from "styled-components";
import SVG from './SVG'
import { media } from "../../Grid";

const Container= styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom:160px;
  ${media['md'](`
  margin-bottom:80px;
  `)};
`;
const Searchbox = styled.div`
  position:relative;
  top:50px;
  background:black;
  height:60px;
  border-radius:40px;
  cursor:pointer;
  transition: width 2s;
  z-index:20;
`;

const SearchInput = styled.input`
  float:left;
  background:none;
  border:none;
  outline:none;
  color:rgb(112, 193, 179);
  font-size: 1em;
  transition: 0.4s;
  padding:0;
  height:60px;
  line-height:60px
  transition: width 0.8s;
  ${Searchbox}:hover & {
    width:200px;
  }
  text-indent: 2em;
  width:0px;
`

const SearchButton = styled.a`
  float:right;
  background: black;
  height:60px;
  width:60px;
  border-radius:50%;
  cursor:pointer;
  transition: background 0.8s;
  ${Searchbox}:hover & {
    background:white;
  }
`;

const SearchBar = ({
        handleInputChange,
        searchTracks
    }) => {
    return (
        <Container>
            <Searchbox>
                <SearchInput
                    placeholder="Search for..."
                    onChange={handleInputChange}
                    onKeyPress={event => { event.key === 'Enter' && searchTracks()}}
                /> 
                <SearchButton onClick={searchTracks}>
                  <SVG fill="rgba(112, 193, 179, 1)" />
                </SearchButton>
            </Searchbox>
        </Container>
    );
};

export default SearchBar;