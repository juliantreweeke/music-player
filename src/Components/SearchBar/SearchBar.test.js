import React from 'react';
import { SearchBar } from './SearchBar';
import { cleanup, render, fireEvent } from '@testing-library/react'

const searchTracks = jest.fn();

describe('SearchBar', () => {
    afterEach(cleanup)

    it("should render", () => {
        const SearchBarComponent = render(
            <SearchBar />
        )
        expect(SearchBarComponent).toBeTruthy();
    });

    it("matches snapshot", () => {
        const SearchBarComponent = render(
            <SearchBar />
        )
        expect(SearchBarComponent).toMatchSnapshot();
    });

    it("calls searchTracks when Search button is clicked", () => {
        const { getByRole } = render(<SearchBar searchTracks={searchTracks} />)
        fireEvent.click(getByRole('button'))
        expect(searchTracks).toHaveBeenCalledTimes(1)
    });
    it("calls searchTracks when enter button is pressed on input", () => {
        const SearchBarComponent = render(
            <SearchBar />
        )
        const input = SearchBarComponent.getByPlaceholderText('Search for...')
        fireEvent.keyDown(input, {
            key: 'Enter',
            keyCode: 13,
        });
        expect(searchTracks).toHaveBeenCalledTimes(1)
    });

    it('should allow letters to be inputted, then used as a parameter for searchTracks function', () => {
        const songQuery = 'my have song bruh';
        const SearchBarComponent = render(
            <SearchBar searchTracks={searchTracks} />
        )
        const { getByRole } = SearchBarComponent;
        const input = SearchBarComponent.getByPlaceholderText('Search for...')
        expect(input.value).toBe('');
        fireEvent.change(input, { target: { value: songQuery } })
        expect(input.value).toBe(songQuery);
        fireEvent.click(getByRole('button'))
        expect(searchTracks).toHaveBeenCalledWith(songQuery);
    })
});
