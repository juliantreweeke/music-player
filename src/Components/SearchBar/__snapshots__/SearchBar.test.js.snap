// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxhCb inopka"
      >
        <div
          class="sc-AxhUy jBaFHR"
        >
          <input
            class="sc-AxgMl hpZkSN"
            placeholder="Search for..."
          />
          <button
            class="sc-AxheI hfWwsZ"
          >
            <svg
              height="100%"
              viewBox="-50 -250 300 700"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <path
                d="M244.186,214.604l-54.379-54.378c-0.289-0.289-0.628-0.491-0.93-0.76 c10.7-16.231,16.945-35.66,16.945-56.554C205.822,46.075,159.747,0,102.911,0S0,46.075,0,102.911 c0,56.835,46.074,102.911,102.91,102.911c20.895,0,40.323-6.245,56.554-16.945c0.269,0.301,0.47,0.64,0.759,0.929l54.38,54.38 c8.169,8.168,21.413,8.168,29.583,0C252.354,236.017,252.354,222.773,244.186,214.604z M102.911,170.146 c-37.134,0-67.236-30.102-67.236-67.235c0-37.134,30.103-67.236,67.236-67.236c37.132,0,67.235,30.103,67.235,67.236 C170.146,140.044,140.043,170.146,102.911,170.146z"
                fill="rgba(112, 193, 179, 1)"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhCb inopka"
    >
      <div
        class="sc-AxhUy jBaFHR"
      >
        <input
          class="sc-AxgMl hpZkSN"
          placeholder="Search for..."
        />
        <button
          class="sc-AxheI hfWwsZ"
        >
          <svg
            height="100%"
            viewBox="-50 -250 300 700"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M244.186,214.604l-54.379-54.378c-0.289-0.289-0.628-0.491-0.93-0.76 c10.7-16.231,16.945-35.66,16.945-56.554C205.822,46.075,159.747,0,102.911,0S0,46.075,0,102.911 c0,56.835,46.074,102.911,102.91,102.911c20.895,0,40.323-6.245,56.554-16.945c0.269,0.301,0.47,0.64,0.759,0.929l54.38,54.38 c8.169,8.168,21.413,8.168,29.583,0C252.354,236.017,252.354,222.773,244.186,214.604z M102.911,170.146 c-37.134,0-67.236-30.102-67.236-67.235c0-37.134,30.103-67.236,67.236-67.236c37.132,0,67.235,30.103,67.235,67.236 C170.146,140.044,140.043,170.146,102.911,170.146z"
              fill="rgba(112, 193, 179, 1)"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
